js/tabRestore.js

```diff
@@ -0,0 +1,33 @@
+const MAX_TAB_HISTORY_DEPTH = 20
+var tabHistoryStack = Array()
```

Comment:
I was thinking about the `tabHistory` and the `tabHistoryStack` variables - one could easily missuse the one instead of the other, and taking in mind that both have a `.push()` method, the distinction between the two could be confusing in another context.
Either way, there are some proposals for module-based system (instead of concatenation) which would solve such problems.